src/app/services/technical-record/technical-record.service.ts

```diff
@@ -62,12 +62,12 @@ export class TechnicalRecordService {
     return this.http.get<VehicleTechRecordModel[]>(url, { responseType: 'json' });
   }
 
-  putUpdateTechRecords(systemNumber: string, techRecord: TechRecordModel, user: { username: string; id?: string }, recordToArchiveStatus?: StatusCodes) {
+  putUpdateTechRecords(systemNumber: string, techRecord: TechRecordModel, user: { username: string; id?: string }, recordToArchiveStatus?: StatusCodes, newStatus?: StatusCodes) {
```

Comment:
Just checking we do actually need both `newStatus` and `recordtoArchive` status? is the `recordToArchiveStatus` presumably the `recordToArchiveStatus` is the old status code?

Reply:
This is done to comply with the API. @aaron-downing, maybe you'd be better equipped to answer this one?

Reply:
So it's because this method is also in charge of setting the status of the new record to be created because we don't change the status in the editable form. 

Previously we could leverage just the `recordToArchiveStatus` parameter but due to us now handling 3 scenarios with this method, we need an additional `newStatus`. For me the method makes more sense as previously setting the new tech record status based on whether `recordToArchiveStatus` was defined seemed off.

The scenarios where these params are passed are:
- Creating a new `provisional` from a `current` record when a `provisional` already exists (recordToArchiveStatus = provisional and newStatus = provisional)
- Promoting a `provisional` record to `current` (recordToArchiveStatus = provisional and newStatus = current).

The scenario where we don't need to pass these params is creating a new provisional from a provisional record when a provisional already exists. The API handles you not sending an `oldStatusCode` param by archiving a record with the status of the record you're trying to create... in this case provisional.